packages/core/types/src/types/core-api/service.ts

```diff
@@ -1,9 +1,11 @@
-import type { Common, Utils } from '..';
+import type { Common, Utils, Entity as E } from '..';
+
+type EntityID = E.ID;
 
 // TODO Use actual entities instead of regular object
-type Entity = { id: string | number } & Record<string, unknown>;
+type Entity = { id: EntityID } & Record<string, unknown>;
```

Comment:
Should I define the entity itself inside the new Entity.ts file instead of here?
cc @Convly

Reply:
Would make a lot of sense. However, this type seems to be very generic & doesn't take into account any model UID or anything. I'm not sure when it has been added or why it's this abstract. Do you know who added it initially to ask more context?



_(pls don't tell me it's me)_

Reply:
![image](https://github.com/strapi/strapi/assets/20578351/3baaae16-e97d-44e2-b2f2-1dd7b329f86e)
saved by the bell

@alexandrebodin could you give us a bit of context of why we are using a generic entity definition here?

Reply:
I've only noticed the todo now. I guess it's part of the migration of strapi/strapi to TS and was aimed to be improved later on?
If so, I don't know if it's worth migrating that right now. I think it can wait for a bigger refactoring. What do you think?

Reply:
I agree 👍 , we will be refactoring it anyway in the v5 branch